server/users.ts

```diff
@@ -307,7 +307,10 @@ type ChatQueueEntry = [string, RoomID, Connection];
 
 export interface UserSettings {
 	blockChallenges: boolean;
-	blockPMs: boolean | AuthLevel;
+	blockPMs: {
```

Comment:
is this the correct type? can someone specify `all` and `specific` (as is currently the case with this type?) is
`{all: boolean | AuthLevel} | {specific?: string[]}` more correct?

Alternatively, `boolean | AuthLevel | string[]` seems like it would work? (ie/ things are specific when theyre an array, otherwise theyre all)

Reply:
Currently this should allow you to specify all (block pms from all users by default not that rank) and allow you to also block specific users - for example, if i did /blockpms autoconfirmed then also added another autoconfirmed user to my blocklist. I feel it'd be nicer to have both at once, but if you can prefer i can remove the hotpatching hack.

Reply:
Oh, I didnt realize it was intended to let you do both. That makes sense, and thus this type signature makes sense. Sorry! This can be resolved :)